### PR TITLE
131 music tab improvements

### LINKED ISSUE #131: improve music tab functionality to include playlists + better q integration w/ spotify

### PR DESCRIPTION
**Pull request by LemonFace0309**
- added functionality to play playlists, q music, and remove music from q.
closes #131

image
image

### CODE REVIEW

File: components/Spaces/CallTabs/Music/Tabs/Queue.js [hunk 1]
```diff
@@ -13,10 +13,32 @@ const useStyles = makeStyles((theme) => ({
...
+    const newQueue = queueClone.reduce((acc, t) => {
+      if (found) return acc.concat([t]);
+      if (t.title === track.title && t.artist === track.artist) {
+        console.debug(t);
+        found = true;
+        return [t];
+      }
+      return [];
+    }, []);
+    setQueue(newQueue);
```
Comment (jimmyygit) [inline components/Spaces/CallTabs/Music/Tabs/Queue.js:30]: maybe comments here.
what is `found` for? Its initially `false` so the first couple songs dont get added until we find a specific track? can we do the same with pop?

File: components/Spaces/CallTabs/Music/Tabs/SearchSongs.js [hunk 2]
```diff
@@ -39,29 +41,43 @@ const SearchSongs = () => {
...
+      newQueue.push(track);
+      return newQueue;
+    });
+    if (newQueue.length <= 1) {
+      setCurrentTrack(track);
+    } else {
+      setNextTracks((prev) => {
+        const newNextTrack = [...prev];
+        newNextTrack.push(track);
+        return newNextTrack;
```
Comment (jimmyygit) [inline components/Spaces/CallTabs/Music/Tabs/SearchSongs.js:78]: cant u do `setNextTracks((prev) => [...prev, track])` ?

**Review by jimmyygit — APPROVED** [review]

File: components/Spaces/CallTabs/Music/Player.js [hunk 3]
```diff
@@ -1,38 +1,64 @@
...
+        (!returnedTracks[1] || queueClone[i + 1]?.title == returnedTracks[1]?.title) &&
+        (!returnedTracks[2] || queueClone[i + 2]?.title == returnedTracks[2]?.title)
+      ) {
+        nextTracksExtened = queueClone.slice(i + 1);
+        break;
+      }
+    }
+    setCurrentTrack(currentTrack);
+    setNextTracks(nextTracksExtened.length > 0 ? nextTracksExtened : nextTracks);
+  };
```
[the remark this answers is not in the document]
Comment (LemonFace0309) [reply components/Spaces/CallTabs/Music/Player.js:46]: true. I'll add documentation. ty Homie

File: components/Spaces/CallTabs/Music/SpotifyProvider.js [hunk 4]
```diff
@@ -30,18 +39,93 @@ export function SpotifyProvider({ children }) {
...
+      .getPlaylistsForCategory('kpop', { limit: 7 })
+      .then((data) => {
+        setKpopPlaylists(() => parsePlaylists(data?.body?.playlists?.items));
+      })
+      .catch((err) => {
+        console.log('Something went wrong!', err);
+      });
+
+    spotifyApi
+      .searchPlaylists('cafe', { limit: 7 })
```
[the remark this answers is not in the document]
Comment (LemonFace0309) [reply components/Spaces/CallTabs/Music/SpotifyProvider.js:76]: probably actually. I'll change it.

File: components/Spaces/CallTabs/Music/Tabs/Queue.js (hunk 1 above)
Comment (LemonFace0309) [reply components/Spaces/CallTabs/Music/Tabs/Queue.js:30]: I'll add documentation.

File: components/Spaces/CallTabs/Music/Player.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (eden-chan) [reply components/Spaces/CallTabs/Music/Player.js:46]: Good suggestion jimmy senpai! 🚀

File: components/Spaces/CallTabs/Music/SpotifyProvider.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (eden-chan) [reply components/Spaces/CallTabs/Music/SpotifyProvider.js:76]: Should these be console.debugs instead? And can the error message be more descriptive

File: components/Spaces/CallTabs/Music/Tabs/SearchSongs.js (hunk 2 above)
Comment (eden-chan) [reply components/Spaces/CallTabs/Music/Tabs/SearchSongs.js:78]: That's actually big brain optimization 🚀

File: components/Spaces/CallTabs/Music/SpotifyProvider.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (LemonFace0309) [reply components/Spaces/CallTabs/Music/SpotifyProvider.js:76]: yes they should ahahaa I miss them too sometimes :D

File: components/Spaces/CallTabs/Music/Tabs/SearchSongs.js (hunk 2 above)
Comment (LemonFace0309) [reply components/Spaces/CallTabs/Music/Tabs/SearchSongs.js]: yes hehe. You reminded me. I actually have to change that function lol.

**Pull request merged**